Sub feat/crud genre song albums

myndaaa opened the pull request:
NOTE: TO BE APPROVED AFTER ` sub-feat/crud-user`
- implemented endpoints of artist, bands, artist band members
- modified schema to remove unused redundant schemas
- added rbac to endpoints
- added new field to `band` s, foreign key to `user` data model to store who is the owner of the `band` via `created_by_user_id` to enforce stricter rbac

myndaaa committed:
Update README.md

myndaaa committed:
Update README.md

myndaaa committed:
added genre and implementations

myndaaa committed:
changes deps location / file path

myndaaa committed:
fixed imports

myndaaa committed:
creation of song crud and api endpoints

myndaaa committed:
admin seed files

myndaaa committed:
docker failed to read .env credentials

myndaaa committed:
dockerization altered to multi stage to reduce image volume

myndaaa committed:
deleted frotnend

myndaaa committed:
deletion of frontend

myndaaa committed:
Merge branch 'sub-feat/crud-artists-bands' into sub-feat/crud-genre-song-albums

myndaaa committed:
Merge branch 'sub-feat/crud-artists-bands' into sub-feat/crud-genre-song-albums

ZulfikarAliZihan commented on backend/app/api/v1/genre.py:
```diff
@@ -0,0 +1,207 @@
+    return genre
+
+
+@router.get("/name/{name}", response_model=GenreOut)
```
Remove this API. Use a query-param to the GET genre API

myndaaa replied:
Pushed updates and changes

ZulfikarAliZihan commented on backend/app/api/v1/genre.py:
```diff
@@ -0,0 +1,207 @@
+    current_admin: dict = Depends(get_current_admin)
+):
+    """Create a new genre - admin only"""
+    if genre_name_taken(db, genre_data.name):
```
Can we do the same thing by only one database call?

myndaaa replied:
can use integrity error to throw an exception, looked it up working similarly to how it works for CREATE ... IF NOT EXISTS in raw sql

ZulfikarAliZihan commented on backend/app/api/v1/song.py:
```diff
@@ -0,0 +1,236 @@
+    return get_all_songs_paginated(db, skip=skip, limit=limit)
+
+
+@router.get("/search", response_model=List[SongOut])
```
Remove this API and add required query-params to the GET API route.

myndaaa committed:
Fix for PR issues:
- genre search is case insensitive
- genre name taken check is case insensitive
- no redundant double GET endpoints for genre, only one get endpoint to get all and get via query param.
- Get endpoint allows close matches
- fixed redundant and useless usage of /admin endpoints to : no deletion option for genre, only disable option., admin see all genre others only active ones
- single db call for genre create, removed prechecks and returning exception error based on crud returns

myndaaa committed:
linter issue fix

myndaaa committed:
- added to do comment on usage of proper rest naming and using names as {name} in endpoints which might have whitespaces.

- added fuzzy search to song.
- rmeoved /search endpoint in song and resuing current get endpoint with query params
